doc/include/api_embedc.inc

```diff
@@ -1,260 +1,191 @@
-this list with custom types if the standard does not support what is
-needed. See section [sectref {Advanced: Extending cproc}] for details.
+[arg resulttype] may be one of the following predefined types or a custom type.
+See section [sectref {Advanced: Extending cproc}].  Unless otherwise noted, the
+Tcl return code is always [const TCL_OK].
 
 [comment {% % %% %%% %%%%% %%%%%%%% %%%%%%%%%%%%% %%%%%%%%%%%%%%%%%%%%%}]
 [include api_stdrt_cproc.inc]
 [comment {% % %% %%% %%%%% %%%%%%%% %%%%%%%%%%%%% %%%%%%%%%%%%%%%%%%%%%}]
 
-[para] The [arg arguments] parameter has the overall syntax of a Tcl
-dictionary value, except that keys (argument names) and values
-(argument types) are specified in reverse order. Consider the
-example [example {
+[para] The [arg arguments] is a multidict where each key is an argument type
+and its value is the argument name.  For example:
+
+[example {
 int x int y
-}] where [arg x] and [arg y] are the arguments/keys, both
-mapped to type/value [const int].
+}]
 
-[para] The argument names must be valid C identifiers.
+[para] Each argument name must be a valid C identifier.
 
-[para] A limited form of variadic arguments is possible, through
-optional arguments with default values. For these the argument name is
-a 2-element list containing the actual name, and the default value.
+[para] If the name is a list containing two items, the first item is the name
+and the second item is the default value.  A limited form of variadic arguments
+can be accomplished using sudh default values.
 
-For example, in the declaration [example { int {x 1} }] [arg x] is an
-optional argument of type [type int] and default value [const 1].
+Example:
 
-[para] A caveat! The default value is assigned unconditionally. If a
-custom argument type uses more complex validation, and the default may
-be invalid according to it, then the relevant checks have to be done
-in the procedure body. The argument conversion cannot do it as it is
-completely bypassed when the argument is not present. Overcoming this
-requires the separation of argument conversion and validation code.
+[example { int {x 1} }]
 
-[para] Below is the list of predefined types legal for arguments, plus
-the details of their semantics. Note that it is possible to extend
-this list with custom types if the standard does not support what is
-needed. See section [sectref {Advanced: Extending cproc}] for details.
+[arg x] is an optional argument of type [type int] with a default value of
+[const 1].
 
-[para] Further note that the type of the first argument is allowed to
-be [const Tcl_Interp*]. In that case the argument in question is
-[emph not] counted as an argument of the new Tcl command.
+[para] Argument conversion is completely bypassed when the argument is not
+provided, so a custom converter doing validation does not get the chance to
+validate the defalut value.  In this case, the value should be checked in the
```

Comment:
`defalut` -> `default`

Reply:
Done with commit 29380521d0873ea1dcc5f8cdff6742b20764473f (Branch `documentation` in my repo, not seen by the PR).